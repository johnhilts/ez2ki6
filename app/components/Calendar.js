import React from 'react';
import { Link } from 'react-router'
import moment from 'moment';

const weekdayHeader = (weekdayName) => {
  return (
    <th key={weekdayName} style={{textAlign: 'center', }}>{weekdayName}</th>
  )
}

// NOTE: I couldn't use the index parameter because what's being mapped is a *filtered* array, not the complete original
// so, to get around it I had to add the index to the object so that I have the original array's index even after going through a filter
const showDate = (monthInfo) => {
  let dayStyle = monthInfo.hasData
    ? {fontWeight: 'bold', fontSize: 'larger', }
    : {fontWeight: 'normal', }
  return (
    monthInfo.isEmpty
    ?
    <td key={monthInfo.absoluteIndex} style={{width: 100, borderStyle: 'dotted', borderWidth: 1,
      paddingLeft: 25, paddingRight: 25, paddingTop: 25, paddingBottom: 25, }}>
      &nbsp;
    </td>
    :
    <td key={monthInfo.absoluteIndex} style={{width: 100, textAlign: 'center', borderStyle: 'solid', borderWidth: 5,
      paddingLeft: 25, paddingRight: 25, paddingTop: 25, paddingBottom: 25, }}>
      <Link style={dayStyle} to={{pathname: 'day', state: {monthInfo: monthInfo, }}}>{monthInfo.day}</Link>
    </td>
  )
}

export default function Calendar(props) {
  return (
    <table cellPadding="10" cellSpacing="10" border="5">
      <thead>
        <tr>
          {moment.weekdays().map(weekdayHeader)}
        </tr>
      </thead>
      <tbody>
        {[1, 2, 3, 4, 5, 6].map(weekIndex => {
          let startCell = (weekIndex - 1) * 7;
          let endCell = startCell + 7;
          return (
            <tr key={weekIndex}>
              {props.monthGrid.filter((monthInfo, cellIndex)=>{return cellIndex >= startCell && cellIndex < endCell;}).map(showDate)}
            </tr>
          )
        }
        )}
      </tbody>
    </table>
  )
}
