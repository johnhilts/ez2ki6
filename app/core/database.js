export const firebaseConfig = {
  apiKey: 'AIzaSyDIoxipbDfUyTsZT_mCj-lShyNOfh_2ROg',
  authDomain: 'easynikkiroku.firebaseapp.com',
  databaseURL: 'https://easynikkiroku.firebaseio.com',
  storageBucket: '',
};

export function getUserRoot(owner) {
  return 'users/' + owner;
}
