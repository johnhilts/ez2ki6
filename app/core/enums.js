export const detailLevel = {day: 0, week: 1, month: 2, month_with_data: 3, };
