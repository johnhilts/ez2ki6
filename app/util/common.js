export const range  = (start, end) => {
  if (typeof(end) != "undefined") {
    return [...Array(end-start).keys()].map(v => start+v)
  }
  else {
    return [...Array(start).keys()]
  }
}
