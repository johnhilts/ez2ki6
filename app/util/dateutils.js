import moment from 'moment';

export const getCurrentFormattedYearMonth = () => {
  return moment().format('YYYYMM');
}

// Input: YYYYMM
// Output: YYYY-MM
const getReformattedYearMonth = (formattedYearMonth) => {
  return formattedYearMonth.substring(0, 4) + '-' + formattedYearMonth.substring(4);
}

// Input: YYYYMM, -1|1
// Output: YYYY-MM
export const getFormattedYearMonthByQueryYearMonth = (queryYearMonth, offset) => {
  let reformattedYearMonth = getReformattedYearMonth(queryYearMonth);
  return moment(reformattedYearMonth).add({months: offset}).format('YYYYMM');
}

// Input: YYYYMM
// Output: moment object
export const getYearMonthByFormattedYearMonth = (formattedYearMonth) => {
  let reformattedYearMonth = getReformattedYearMonth(formattedYearMonth);
  return moment(reformattedYearMonth);
}

// Input: y, m
// Output: YYYYMM
export const getFormattedYearMonthByYearMonth = (year, month) => {
  return moment(new Date(year, month, 1)).format('YYYYMM');
}

// Input: m
// Output: MMMM
export const getMonthName = (month) => {
  return moment(new Date(moment().year(), month, 1)).format('MMMM');
}

// returns date object based on default moment object
export const getCurrentDate = () => {
  return {year: moment().year(), month: moment().month(), day: moment().date(), };
}

// Input: y,m,d
// Output: true or false if y,m,d is for the actual current date
export const isToday = (year, month, day) => {
  let date = moment(new Date(year, month, day));
  let today = moment(new Date());
  return today.isSame(date, 'day');
}

// Input: y,m,d
// Output: date object
export const getDateFromYearMonthDay = (year, month, day) => {
  let date = moment(new Date(year, month, day));
  return {year: date.year(), month: date.month(), day: date.date(), };
}

// Input: y,m,d
// Output: MM/DD/YYYY
export const getFormattedMonthDayYear = (year, month, day) => {
  return moment(new Date(year, month, day)).format('MM/DD/YYYY');
}

export const getDaysByYearMonth = (year, month) => {
  return moment(new Date(year, month, 1)).daysInMonth();
}
