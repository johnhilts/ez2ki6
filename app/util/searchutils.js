export const searchByDateRange = (searchText, from, to, date) => {
    let fromDate = new Date(from.year, from.month, from.day);
    let toDate = new Date(to.year, to.month, to.day);
    let compareDate = new Date(date.year, date.month, date.day);
    return (
        date.dateInfo.toLowerCase().indexOf(searchText.toLowerCase()) >= 0
            && fromDate <= compareDate
            && toDate >= compareDate
    );
}