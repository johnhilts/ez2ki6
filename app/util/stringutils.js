import React from 'react';

export const formatLineBreaksForHtml = (str) => {
	return str.split("\n").map(function(item) {
		return (
			<span>
			{item}
			<br/>
			</span>
		       )
	});
}
