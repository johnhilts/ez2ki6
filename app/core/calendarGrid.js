import * as dateUtils from '../util/dateutils';
import * as enums from '../core/enums';

const saturday = 6;

export const getDayInfo = (dates, currentDate, isCurrentMonth, absoluteCellIndex) => {

  const matchesDate = dbDate => {return dbDate.year == currentDate.year && dbDate.month == currentDate.month && dbDate.day == currentDate.day;};

  let isToday = dateUtils.isToday(currentDate.year, currentDate.month, currentDate.day);
  let {year, month, day} = currentDate;
  let dayInfo = {year, month, day, absoluteIndex: absoluteCellIndex, isToday: isToday, isCurrentMonth: isCurrentMonth, };
  dayInfo.hasData = dates.some(matchesDate);
  if (dayInfo.hasData) {
    dayInfo.dateEntries = dates.filter(matchesDate);
  }

  return dayInfo;
}

export const getLastCalendarCellIndexFromStartOfCurrentMonth = (year, month, lastDay) => {
  let lastCurrentMonthWeekday = dateUtils.getWeekdayFromYearMonthDay(year, month, lastDay);
  return (
    lastCurrentMonthWeekday == saturday
    ? lastDay
    : lastDay + (saturday - lastCurrentMonthWeekday)
  )
}

export const buildMonthGrid = (dates, currentYearMonth) => {

  const getDayCellInfo = (currentDate, isCurrentMonth) => {
    return getDayInfo(dates, currentDate, isCurrentMonth, absoluteCellIndex++);
  }

  const daysInMonth = currentYearMonth.daysInMonth;
  let monthGrid = [];
  let absoluteCellIndex = 0;
  const currentYear = currentYearMonth.year;
  const currentMonth = currentYearMonth.month;
  const lastCalendarCellIndex = getLastCalendarCellIndexFromStartOfCurrentMonth(currentYear, currentMonth, daysInMonth);

  const processCalendarCell = (cellIndex) => {
    let day = cellIndex + 1;
    let date = dateUtils.getDateFromYearMonthDay(currentYear, currentMonth, day);

    const partOfWeekInPreviousMonth = (cellIndex) => {
      let previousMonthDate = dateUtils.addDays(currentYear, currentMonth, day, (weekday - cellIndex) * -1);
      return getDayCellInfo(previousMonthDate, false);
    }

    const weekInCurrentMonth = (cellIndex) => {
      return getDayCellInfo(date, true);
    }

    const partOfWeekInNextMonth = (cellIndex) => {
      return getDayCellInfo(date, false);
    }

    const daysInSameWeekPreviousMonth = (weekCellIndex) => {
      return day == 1 && weekday > weekCellIndex && weekCellIndex < saturday;
    }

    const daysInCurrentMonth = (weekCellIndex) => {
      return weekday == weekCellIndex && day <= daysInMonth;
    }

    const daysInSameWeekNextMonth = (weekCellIndex) => {
      return weekday == weekCellIndex && day > daysInMonth;
    }

    const weekLength = 7;
    let calendarCellsForWeek = [...Array(weekLength).keys()];
    let weekday = dateUtils.getWeekdayFromYearMonthDay(currentYear, currentMonth, day);

    monthGrid.push(...calendarCellsForWeek.filter(daysInSameWeekPreviousMonth).map(partOfWeekInPreviousMonth));
    monthGrid.push(...calendarCellsForWeek.filter(daysInCurrentMonth).map(weekInCurrentMonth));
    monthGrid.push(...calendarCellsForWeek.filter(daysInSameWeekNextMonth).map(partOfWeekInNextMonth));
  }

  let calendarCellsFromStartOfCurrentMonth = [...Array(lastCalendarCellIndex).keys()];
  calendarCellsFromStartOfCurrentMonth.map(processCalendarCell);

  return monthGrid;
}

export const setLinkDetailLevelForMonthView = (monthGrid) => {
  if (monthGrid.some(m => m.hasData)) {
    return enums.detailLevel.month_with_data
  }
  else {
    return enums.detailLevel.month
  }
}
