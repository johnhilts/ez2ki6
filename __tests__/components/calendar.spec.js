import React, { Component } from 'react';
import ReactTestRenderer from 'react-test-renderer';
import Calendar from '../../app/components/Calendar';
import * as enums from '../../app/core/enums';
import * as calendarGrid from '../../app/core/calendarGrid';
import * as dateUtils from '../../app/util/dateutils';
import * as testHelper from '../../test/Helper/DateHelper';

describe('Calendar component snapshot test', () => {

  it('Should compare the Calendar component with a snapshot', () => {
    const component = ReactTestRenderer.create(
      <Calendar monthGrid={calendarGrid.buildMonthGrid(testHelper.dates, testHelper.currentYearMonth)} />
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  })
});
