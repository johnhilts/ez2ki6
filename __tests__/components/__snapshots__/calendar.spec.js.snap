exports[`Calendar component snapshot test Should compare the Calendar component with a snapshot 1`] = `
<table
  cellPadding="10"
  cellSpacing="10">
  <thead>
    <tr>
      <th
        style={
          Object {
            "textAlign": "center",
          }
        }>
        Sunday
      </th>
      <th
        style={
          Object {
            "textAlign": "center",
          }
        }>
        Monday
      </th>
      <th
        style={
          Object {
            "textAlign": "center",
          }
        }>
        Tuesday
      </th>
      <th
        style={
          Object {
            "textAlign": "center",
          }
        }>
        Wednesday
      </th>
      <th
        style={
          Object {
            "textAlign": "center",
          }
        }>
        Thursday
      </th>
      <th
        style={
          Object {
            "textAlign": "center",
          }
        }>
        Friday
      </th>
      <th
        style={
          Object {
            "textAlign": "center",
          }
        }>
        Saturday
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td
        className="gridLines"
        style={
          Object {
            "borderStyle": "solid",
            "borderWidth": 5,
            "paddingBottom": 25,
            "paddingLeft": 25,
            "paddingRight": 25,
            "paddingTop": 25,
            "textAlign": "center",
            "verticalAlign": "middle",
            "width": 100,
          }
        }>
        <a
          onClick={[Function]}
          style={Object {}}>
          <div
            style={
              Object {
                "fontSize": "larger",
                "fontWeight": "bold",
              }
            }>
            1
            <span
              className="glyphicon glyphicon-list-alt" />
            
          </div>
        </a>
      </td>
      <td
        className="gridLines"
        style={
          Object {
            "borderStyle": "solid",
            "borderWidth": 5,
            "paddingBottom": 25,
            "paddingLeft": 25,
            "paddingRight": 25,
            "paddingTop": 25,
            "textAlign": "center",
            "verticalAlign": "middle",
            "width": 100,
          }
        }>
        <a
          onClick={[Function]}
          style={Object {}}>
          <div
            style={
              Object {
                "fontSize": "larger",
                "fontWeight": "bold",
              }
            }>
            2
            <span
              className="glyphicon glyphicon-list-alt" />
            
          </div>
        </a>
      </td>
      <td
        className="gridLines"
        style={
          Object {
            "borderStyle": "solid",
            "borderWidth": 5,
            "paddingBottom": 25,
            "paddingLeft": 25,
            "paddingRight": 25,
            "paddingTop": 25,
            "textAlign": "center",
            "verticalAlign": "middle",
            "width": 100,
          }
        }>
        <a
          onClick={[Function]}
          style={Object {}}>
          <div
            style={
              Object {
                "fontWeight": "normal",
              }
            }>
            3
            
            
          </div>
        </a>
      </td>
      <td
        className="gridLines"
        style={
          Object {
            "borderStyle": "solid",
            "borderWidth": 5,
            "paddingBottom": 25,
            "paddingLeft": 25,
            "paddingRight": 25,
            "paddingTop": 25,
            "textAlign": "center",
            "verticalAlign": "middle",
            "width": 100,
          }
        }>
        <a
          onClick={[Function]}
          style={Object {}}>
          <div
            style={
              Object {
                "fontWeight": "normal",
              }
            }>
            4
            
            
          </div>
        </a>
      </td>
      <td
        className="gridLines"
        style={
          Object {
            "borderStyle": "solid",
            "borderWidth": 5,
            "paddingBottom": 25,
            "paddingLeft": 25,
            "paddingRight": 25,
            "paddingTop": 25,
            "textAlign": "center",
            "verticalAlign": "middle",
            "width": 100,
          }
        }>
        <a
          onClick={[Function]}
          style={Object {}}>
          <div
            style={
              Object {
                "fontWeight": "normal",
              }
            }>
            5
            
            
          </div>
        </a>
      </td>
      <td
        className="gridLines"
        style={
          Object {
            "borderStyle": "solid",
            "borderWidth": 5,
            "paddingBottom": 25,
            "paddingLeft": 25,
            "paddingRight": 25,
            "paddingTop": 25,
            "textAlign": "center",
            "verticalAlign": "middle",
            "width": 100,
          }
        }>
        <a
          onClick={[Function]}
          style={Object {}}>
          <div
            style={
              Object {
                "fontWeight": "normal",
              }
            }>
            6
            
            
          </div>
        </a>
      </td>
      <td
        className="gridLines"
        style={
          Object {
            "borderStyle": "solid",
            "borderWidth": 5,
            "paddingBottom": 25,
            "paddingLeft": 25,
            "paddingRight": 25,
            "paddingTop": 25,
            "textAlign": "center",
            "verticalAlign": "middle",
            "width": 100,
          }
        }>
        <a
          onClick={[Function]}
          style={Object {}}>
          <div
            style={
              Object {
                "fontWeight": "normal",
              }
            }>
            7
            
            
          </div>
        </a>
      </td>
    </tr>
    <tr>
      <td
        className="gridLines"
        style={
          Object {
            "borderStyle": "solid",
            "borderWidth": 5,
            "paddingBottom": 25,
            "paddingLeft": 25,
            "paddingRight": 25,
            "paddingTop": 25,
            "textAlign": "center",
            "verticalAlign": "middle",
            "width": 100,
          }
        }>
        <a
          onClick={[Function]}
          style={Object {}}>
          <div
            style={
              Object {
                "fontWeight": "normal",
              }
            }>
            8
            
            
          </div>
        </a>
      </td>
      <td
        className="gridLines"
        style={
          Object {
            "borderStyle": "solid",
            "borderWidth": 5,
            "paddingBottom": 25,
            "paddingLeft": 25,
            "paddingRight": 25,
            "paddingTop": 25,
            "textAlign": "center",
            "verticalAlign": "middle",
            "width": 100,
          }
        }>
        <a
          onClick={[Function]}
          style={Object {}}>
          <div
            style={
              Object {
                "fontWeight": "normal",
              }
            }>
            9
            
            
          </div>
        </a>
      </td>
      <td
        className="gridLines"
        style={
          Object {
            "borderStyle": "solid",
            "borderWidth": 5,
            "paddingBottom": 25,
            "paddingLeft": 25,
            "paddingRight": 25,
            "paddingTop": 25,
            "textAlign": "center",
            "verticalAlign": "middle",
            "width": 100,
          }
        }>
        <a
          onClick={[Function]}
          style={Object {}}>
          <div
            style={
              Object {
                "fontWeight": "normal",
              }
            }>
            10
            
            
          </div>
        </a>
      </td>
      <td
        className="gridLines"
        style={
          Object {
            "borderStyle": "solid",
            "borderWidth": 5,
            "paddingBottom": 25,
            "paddingLeft": 25,
            "paddingRight": 25,
            "paddingTop": 25,
            "textAlign": "center",
            "verticalAlign": "middle",
            "width": 100,
          }
        }>
        <a
          onClick={[Function]}
          style={Object {}}>
          <div
            style={
              Object {
                "fontWeight": "normal",
              }
            }>
            11
            
            
          </div>
        </a>
      </td>
      <td
        className="gridLines"
        style={
          Object {
            "borderStyle": "solid",
            "borderWidth": 5,
            "paddingBottom": 25,
            "paddingLeft": 25,
            "paddingRight": 25,
            "paddingTop": 25,
            "textAlign": "center",
            "verticalAlign": "middle",
            "width": 100,
          }
        }>
        <a
          onClick={[Function]}
          style={Object {}}>
          <div
            style={
              Object {
                "fontWeight": "normal",
              }
            }>
            12
            
            
          </div>
        </a>
      </td>
      <td
        className="gridLines"
        style={
          Object {
            "borderStyle": "solid",
            "borderWidth": 5,
            "paddingBottom": 25,
            "paddingLeft": 25,
            "paddingRight": 25,
            "paddingTop": 25,
            "textAlign": "center",
            "verticalAlign": "middle",
            "width": 100,
          }
        }>
        <a
          onClick={[Function]}
          style={Object {}}>
          <div
            style={
              Object {
                "fontWeight": "normal",
              }
            }>
            13
            
            
          </div>
        </a>
      </td>
      <td
        className="gridLines"
        style={
          Object {
            "borderStyle": "solid",
            "borderWidth": 5,
            "paddingBottom": 25,
            "paddingLeft": 25,
            "paddingRight": 25,
            "paddingTop": 25,
            "textAlign": "center",
            "verticalAlign": "middle",
            "width": 100,
          }
        }>
        <a
          onClick={[Function]}
          style={Object {}}>
          <div
            style={
              Object {
                "fontWeight": "normal",
              }
            }>
            14
            
            
          </div>
        </a>
      </td>
    </tr>
    <tr>
      <td
        className="gridLines"
        style={
          Object {
            "borderStyle": "solid",
            "borderWidth": 5,
            "paddingBottom": 25,
            "paddingLeft": 25,
            "paddingRight": 25,
            "paddingTop": 25,
            "textAlign": "center",
            "verticalAlign": "middle",
            "width": 100,
          }
        }>
        <a
          onClick={[Function]}
          style={Object {}}>
          <div
            style={
              Object {
                "fontWeight": "normal",
              }
            }>
            15
            
            
          </div>
        </a>
      </td>
      <td
        className="gridLines"
        style={
          Object {
            "borderStyle": "solid",
            "borderWidth": 5,
            "paddingBottom": 25,
            "paddingLeft": 25,
            "paddingRight": 25,
            "paddingTop": 25,
            "textAlign": "center",
            "verticalAlign": "middle",
            "width": 100,
          }
        }>
        <a
          onClick={[Function]}
          style={Object {}}>
          <div
            style={
              Object {
                "fontWeight": "normal",
              }
            }>
            16
            
            
          </div>
        </a>
      </td>
      <td
        className="gridLines"
        style={
          Object {
            "borderStyle": "solid",
            "borderWidth": 5,
            "paddingBottom": 25,
            "paddingLeft": 25,
            "paddingRight": 25,
            "paddingTop": 25,
            "textAlign": "center",
            "verticalAlign": "middle",
            "width": 100,
          }
        }>
        <a
          onClick={[Function]}
          style={Object {}}>
          <div
            style={
              Object {
                "fontWeight": "normal",
              }
            }>
            17
            
            
          </div>
        </a>
      </td>
      <td
        className="gridLines"
        style={
          Object {
            "borderStyle": "solid",
            "borderWidth": 5,
            "paddingBottom": 25,
            "paddingLeft": 25,
            "paddingRight": 25,
            "paddingTop": 25,
            "textAlign": "center",
            "verticalAlign": "middle",
            "width": 100,
          }
        }>
        <a
          onClick={[Function]}
          style={Object {}}>
          <div
            style={
              Object {
                "fontWeight": "normal",
              }
            }>
            18
            
            
          </div>
        </a>
      </td>
      <td
        className="gridLines"
        style={
          Object {
            "borderStyle": "solid",
            "borderWidth": 5,
            "paddingBottom": 25,
            "paddingLeft": 25,
            "paddingRight": 25,
            "paddingTop": 25,
            "textAlign": "center",
            "verticalAlign": "middle",
            "width": 100,
          }
        }>
        <a
          onClick={[Function]}
          style={Object {}}>
          <div
            style={
              Object {
                "fontWeight": "normal",
              }
            }>
            19
            
            
          </div>
        </a>
      </td>
      <td
        className="gridLines"
        style={
          Object {
            "borderStyle": "solid",
            "borderWidth": 5,
            "paddingBottom": 25,
            "paddingLeft": 25,
            "paddingRight": 25,
            "paddingTop": 25,
            "textAlign": "center",
            "verticalAlign": "middle",
            "width": 100,
          }
        }>
        <a
          onClick={[Function]}
          style={Object {}}>
          <div
            style={
              Object {
                "fontWeight": "normal",
              }
            }>
            20
            
            
          </div>
        </a>
      </td>
      <td
        className="gridLines"
        style={
          Object {
            "borderStyle": "solid",
            "borderWidth": 5,
            "paddingBottom": 25,
            "paddingLeft": 25,
            "paddingRight": 25,
            "paddingTop": 25,
            "textAlign": "center",
            "verticalAlign": "middle",
            "width": 100,
          }
        }>
        <a
          onClick={[Function]}
          style={Object {}}>
          <div
            style={
              Object {
                "fontWeight": "normal",
              }
            }>
            21
            
            
          </div>
        </a>
      </td>
    </tr>
    <tr>
      <td
        className="gridLines"
        style={
          Object {
            "borderStyle": "solid",
            "borderWidth": 5,
            "paddingBottom": 25,
            "paddingLeft": 25,
            "paddingRight": 25,
            "paddingTop": 25,
            "textAlign": "center",
            "verticalAlign": "middle",
            "width": 100,
          }
        }>
        <a
          onClick={[Function]}
          style={Object {}}>
          <div
            style={
              Object {
                "fontWeight": "normal",
              }
            }>
            22
            
            
          </div>
        </a>
      </td>
      <td
        className="gridLines"
        style={
          Object {
            "borderStyle": "solid",
            "borderWidth": 5,
            "paddingBottom": 25,
            "paddingLeft": 25,
            "paddingRight": 25,
            "paddingTop": 25,
            "textAlign": "center",
            "verticalAlign": "middle",
            "width": 100,
          }
        }>
        <a
          onClick={[Function]}
          style={Object {}}>
          <div
            style={
              Object {
                "fontWeight": "normal",
              }
            }>
            23
            
            
          </div>
        </a>
      </td>
      <td
        className="gridLines"
        style={
          Object {
            "borderStyle": "solid",
            "borderWidth": 5,
            "paddingBottom": 25,
            "paddingLeft": 25,
            "paddingRight": 25,
            "paddingTop": 25,
            "textAlign": "center",
            "verticalAlign": "middle",
            "width": 100,
          }
        }>
        <a
          onClick={[Function]}
          style={Object {}}>
          <div
            style={
              Object {
                "fontWeight": "normal",
              }
            }>
            24
            
            
          </div>
        </a>
      </td>
      <td
        className="gridLines"
        style={
          Object {
            "borderStyle": "solid",
            "borderWidth": 5,
            "paddingBottom": 25,
            "paddingLeft": 25,
            "paddingRight": 25,
            "paddingTop": 25,
            "textAlign": "center",
            "verticalAlign": "middle",
            "width": 100,
          }
        }>
        <a
          onClick={[Function]}
          style={Object {}}>
          <div
            style={
              Object {
                "fontWeight": "normal",
              }
            }>
            25
            
            
          </div>
        </a>
      </td>
      <td
        className="gridLines"
        style={
          Object {
            "borderStyle": "solid",
            "borderWidth": 5,
            "paddingBottom": 25,
            "paddingLeft": 25,
            "paddingRight": 25,
            "paddingTop": 25,
            "textAlign": "center",
            "verticalAlign": "middle",
            "width": 100,
          }
        }>
        <a
          onClick={[Function]}
          style={Object {}}>
          <div
            style={
              Object {
                "fontWeight": "normal",
              }
            }>
            26
            
            
          </div>
        </a>
      </td>
      <td
        className="gridLines"
        style={
          Object {
            "borderStyle": "solid",
            "borderWidth": 5,
            "paddingBottom": 25,
            "paddingLeft": 25,
            "paddingRight": 25,
            "paddingTop": 25,
            "textAlign": "center",
            "verticalAlign": "middle",
            "width": 100,
          }
        }>
        <a
          onClick={[Function]}
          style={Object {}}>
          <div
            style={
              Object {
                "fontWeight": "normal",
              }
            }>
            27
            
            
          </div>
        </a>
      </td>
      <td
        className="gridLines"
        style={
          Object {
            "borderStyle": "solid",
            "borderWidth": 5,
            "paddingBottom": 25,
            "paddingLeft": 25,
            "paddingRight": 25,
            "paddingTop": 25,
            "textAlign": "center",
            "verticalAlign": "middle",
            "width": 100,
          }
        }>
        <a
          onClick={[Function]}
          style={Object {}}>
          <div
            style={
              Object {
                "fontWeight": "normal",
              }
            }>
            28
            
            
          </div>
        </a>
      </td>
    </tr>
    <tr>
      <td
        className="gridLines"
        style={
          Object {
            "borderStyle": "solid",
            "borderWidth": 5,
            "paddingBottom": 25,
            "paddingLeft": 25,
            "paddingRight": 25,
            "paddingTop": 25,
            "textAlign": "center",
            "verticalAlign": "middle",
            "width": 100,
          }
        }>
        <a
          onClick={[Function]}
          style={Object {}}>
          <div
            style={
              Object {
                "fontWeight": "normal",
              }
            }>
            29
            
            
          </div>
        </a>
      </td>
      <td
        className="gridLines"
        style={
          Object {
            "borderStyle": "solid",
            "borderWidth": 5,
            "paddingBottom": 25,
            "paddingLeft": 25,
            "paddingRight": 25,
            "paddingTop": 25,
            "textAlign": "center",
            "verticalAlign": "middle",
            "width": 100,
          }
        }>
        <a
          onClick={[Function]}
          style={Object {}}>
          <div
            style={
              Object {
                "fontWeight": "normal",
              }
            }>
            30
            
            
          </div>
        </a>
      </td>
      <td
        className="gridLines"
        style={
          Object {
            "borderStyle": "solid",
            "borderWidth": 5,
            "paddingBottom": 25,
            "paddingLeft": 25,
            "paddingRight": 25,
            "paddingTop": 25,
            "textAlign": "center",
            "verticalAlign": "middle",
            "width": 100,
          }
        }>
        <a
          onClick={[Function]}
          style={Object {}}>
          <div
            style={
              Object {
                "fontWeight": "normal",
              }
            }>
            31
            
            
          </div>
        </a>
      </td>
      <td
        className="gridLines"
        style={
          Object {
            "borderStyle": "dotted",
            "borderWidth": 1,
            "paddingBottom": 25,
            "paddingLeft": 25,
            "paddingRight": 25,
            "paddingTop": 25,
            "textAlign": "center",
            "verticalAlign": "middle",
            "width": 100,
          }
        }>
        <a
          onClick={[Function]}
          style={Object {}}>
          <div
            style={
              Object {
                "fontWeight": "normal",
              }
            }>
            1
            
            
          </div>
        </a>
      </td>
      <td
        className="gridLines"
        style={
          Object {
            "borderStyle": "dotted",
            "borderWidth": 1,
            "paddingBottom": 25,
            "paddingLeft": 25,
            "paddingRight": 25,
            "paddingTop": 25,
            "textAlign": "center",
            "verticalAlign": "middle",
            "width": 100,
          }
        }>
        <a
          onClick={[Function]}
          style={Object {}}>
          <div
            style={
              Object {
                "fontWeight": "normal",
              }
            }>
            2
            
            
          </div>
        </a>
      </td>
      <td
        className="gridLines"
        style={
          Object {
            "borderStyle": "dotted",
            "borderWidth": 1,
            "paddingBottom": 25,
            "paddingLeft": 25,
            "paddingRight": 25,
            "paddingTop": 25,
            "textAlign": "center",
            "verticalAlign": "middle",
            "width": 100,
          }
        }>
        <a
          onClick={[Function]}
          style={Object {}}>
          <div
            style={
              Object {
                "fontWeight": "normal",
              }
            }>
            3
            
            
          </div>
        </a>
      </td>
      <td
        className="gridLines"
        style={
          Object {
            "borderStyle": "dotted",
            "borderWidth": 1,
            "paddingBottom": 25,
            "paddingLeft": 25,
            "paddingRight": 25,
            "paddingTop": 25,
            "textAlign": "center",
            "verticalAlign": "middle",
            "width": 100,
          }
        }>
        <a
          onClick={[Function]}
          style={Object {}}>
          <div
            style={
              Object {
                "fontWeight": "normal",
              }
            }>
            4
            
            
          </div>
        </a>
      </td>
    </tr>
    <tr />
  </tbody>
</table>
`;
